lib/rules/prefer-const.js

```diff
@@ -240,11 +259,32 @@ module.exports = {
          * @returns {void}
          */
         function report(node) {
-            context.report({
-                node: node,
-                message: "'{{name}}' is never reassigned, use 'const' instead.",
-                data: node
-            });
+            var reportArgs = {
+                    node: node,
+                    message: "'{{name}}' is never reassigned, use 'const' instead.",
+                    data: node
+                },
+                isNormalVariableDeclaration = node.parent.parent.type === "VariableDeclaration" &&
+                    (node.parent.parent.parent.type === "ForInStatement" ||
+                        node.parent.parent.parent.type === "ForOfStatement" ||
+                        node.parent.parent.type === "ForStatement" ||
+                        node.parent.init),
+                isDestructuring = node.parent.parent.type === "ObjectPattern" &&
+                    node.parent.parent.parent.parent.type === "VariableDeclaration",
+                variableDeclarationParent = findUp(node, "VariableDeclaration");
+
+            if (isNormalVariableDeclaration || isDestructuring) {
+
+                reportArgs.fix = function(fixer) {
```

Comment:
Are you talking about a case like this?

```js
let {a, b} = obj; // b should be const
a = 1;
```

I was going to leave this case unfixed, because it's not obvious how a developer would want to resolve the warning.